nomad/structs/structs.go

```diff
@@ -7774,6 +7776,88 @@ func (tg *TaskGroup) SetConstraints(newConstraints []*Constraint) {
 	tg.Constraints = newConstraints
 }
 
+// TaskGroupVolumeClaim associates a task group with a host volume ID. It's
+// used for stateful deployments, i.e., volume requests with "sticky" set to
+// true.
+type TaskGroupVolumeClaim struct {
+	Namespace     string
+	JobID         string
+	TaskGroupName string
+	AllocID       string // used for checks to make sure we don't insert duplicate claims for the same alloc
+
+	VolumeID   string
+	VolumeName string
+
+	// Hash is the hashed value of the claim and is generated using all fields
+	// from the full object except the create and modify times and indexes.
+	Hash []byte
+
+	CreateTime  time.Time
+	ModifyTime  time.Time
+	CreateIndex uint64
+	ModifyIndex uint64
+}
+
+// SetHash is used to compute and set the hash of the task group volume claim.
+// This should be called every time a user specified field on the method is
+// changed before updating the Nomad state store.
+func (tgvc *TaskGroupVolumeClaim) SetHash() []byte {
```

Comment:
Doesn't look like we use this hash. _Usually_ we only have that for when comparisons are expensive and we're replicating an object between regions (ex. ACL policies).

Reply:
eh yeah this was copy pasta from some other state store object and I used it for equality checks between claims. But we don't really need it, I'll clean this up.